provider/datagen/src/transform/cldr/cldr_serde/currencies.rs

```diff
@@ -0,0 +1,40 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Serde structs representing CLDR JSON numbers.json files.
+//!
+//! Sample file:
+//! <https://github.com/unicode-org/cldr-json/blob/master/cldr-json/cldr-numbers-full/main/en/currencies.json>
+
+use icu_locid::LanguageIdentifier;
+use serde::Deserialize;
+use std::collections::{BTreeMap, HashMap};
+use tinystr::UnvalidatedTinyAsciiStr;
+
+#[derive(PartialEq, Debug, Deserialize)]
+pub struct CurrencyPatterns {
+    #[serde(rename = "symbol")]
+    pub short: Option<String>,
+
+    #[serde(rename = "symbol-alt-narrow")]
+    pub narrow: Option<String>,
+}
+
+#[derive(PartialEq, Debug, Deserialize)]
+pub struct Numbers {
+    pub currencies: BTreeMap<UnvalidatedTinyAsciiStr<3>, CurrencyPatterns>,
+}
+
+#[derive(PartialEq, Debug, Deserialize)]
+pub struct LangNumbers {
+    pub numbers: Numbers,
+}
+
+#[derive(PartialEq, Debug, Deserialize)]
+pub struct LangData(pub HashMap<LanguageIdentifier, LangNumbers>);
+
+#[derive(PartialEq, Debug, Deserialize)]
+pub struct Resource {
+    pub main: LangData,
+}
```

Comment:
Shane recently added a new to handle the top levels, see https://github.com/unicode-org/icu4x/pull/3768

Reply:
yeah, I saw it, it broke my implementation in the beginning and I fixed it.

Reply:
Please use the new `LocaleResource` type. Your code will break soon if you aren't using it.